### PR TITLE
added bbox class method to regular mesh

### LINKED ISSUE #2210: feature request: RegularMesh from universe class method
I find myself making RegularMesh that encompass the entire geometry quite often. When making weight windows, flux maps or dose maps.

We have ```Universe.bounding_box``` property and I was wondering if anyone else would like a new class method on the ```RegularMesh``` class so that regular meshes can be made from a universe.

This would automatically set ```lower_left``` and the ```upper_right``` using the universe.bounding_box

Perhaps not a huge saving for the user but let me know if this is a worthwhile upgrade and I can make a PR

usage
```python
mesh = openmc.RegularMesh.from_universe(my_universe, dimension=[100, 100, 100])
```

current method
```python
mesh = openmc.RegularMesh()
mesh.dimension = [100, 100, 100]
mesh.lower_left = geometry.bounding_box[0]
mesh.upper_right = geometry.bounding_box[1]
```

additional code required in mesh.py
```python
    @classmethod
    def from_universe(cls, universe, dimension=[10, 10, 10], mesh_id=None, name=''):
        """Create mesh from an existing openmc universe

        Parameters
        ----------
        universe : openmc.Universe
            Open used as a template for this mesh
        dimension : Iterable of int
            The number of mesh cells in each direction.
        mesh_id : int
            Unique identifier for the mesh
        name : str
            Name of the mesh

        Returns
        -------
        openmc.RegularMesh
            RegularMesh instance

        """
        cv.check_type('rectangular lattice', universe, openmc.Universe)

        mesh = cls(mesh_id, name)
        mesh.lower_left = universe.bounding_box[0]
        mesh.upper_right = universe.bounding_box[1]
        mesh.dimension = dimension

        return mesh
```

### PR DESCRIPTION
**Pull request by shimwell**
This PR is an attempt to add a class method to ```RegularMesh``` that allows construction of a regular mesh from a cell, region, universe or geometry.

Merging this PR closes #2210

### CODE REVIEW

File: openmc/mesh.py [hunk 1]
```diff
@@ -509,6 +510,49 @@ def from_rect_lattice(cls, lattice, division=1, mesh_id=None, name=''):
 
         return mesh
 
+    @classmethod
+    def from_bounding_box(
+        cls,
+        bounding_box,
```
Comment (pshriwise) [inline openmc/mesh.py:516]: I'll suggest replacing 'bounding_box' with 'domain' throughout to capture what these incoming objects represent.

Comment (shimwell) [reply openmc/mesh.py:516]: ah yes that is a nice catch all term to use for cell, region, geometry, universe.

File: openmc/mesh.py [hunk 2]
```diff
@@ -509,6 +510,49 @@ def from_rect_lattice(cls, lattice, division=1, mesh_id=None, name=''):
...
+        name=''
+    ):
+        """Create mesh from an existing openmc cell, region, universe or
+        geometry by making use of the objects bounding box property.
+
+        Parameters
+        ----------
+        domain : {openmc.Cell, openmc.Region, openmc.Universe, openmc.Geometry}
+            The object passed in will be used as a template for this mesh. The
+            domain of the property of the object passed will be used to
```
Comment (shimwell) [inline openmc/mesh.py:528]:
```suggestion
            bounding box of the property of the object passed will be used to
```

File: tests/unit_tests/test_mesh_from_domain.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import numpy as np
+import openmc
+import pytest
+
+
+def test_mesh_from_cell():
```
Comment (pshriwise) [inline tests/unit_tests/test_mesh_from_domain.py:6]: How would you feel about naming these `test_reg_mesh_*`? I could see us adding support this kind of method on other mesh types in the future.

File: tests/unit_tests/test_mesh_from_domain.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    geometry = openmc.Geometry(universe)
+
+    mesh = openmc.RegularMesh.from_domain(geometry)
+    assert isinstance(mesh, openmc.RegularMesh)
+    assert np.array_equal(mesh.dimension, (100, 100, 100))  # default values
+    assert np.array_equal(mesh.lower_left, geometry.bounding_box[0])
+    assert np.array_equal(mesh.upper_right, geometry.bounding_box[1])
+
+
+def test_error_from_unsupported_object():
```
Comment (pshriwise) [inline tests/unit_tests/test_mesh_from_domain.py:61]: Nice 😄

**Review by pshriwise — COMMENTED** [review]
One more thought, but it looks pretty good to me!

File: tests/unit_tests/test_mesh_from_domain.py (hunk 3 above)
Comment (shimwell) [reply tests/unit_tests/test_mesh_from_domain.py]: that is a better name, thanks for the suggestion.

**Review by pshriwise — APPROVED** [review]
I'll give @paulromano some time to take a look before merging, but this looks good to me.

**Review by paulromano — APPROVED** [review]

**Pull request merged**